### ISSUE TITLE
xDrip Missed Reading Issues 

### ISSUE THREAD

**Issue by bjblackmore** [NightscoutFoundation/xDrip#1874]
I had an update to xDrip, version 49817a5-2021.10.04 code 2110041402, from 17cbcf0-2021.03.08 and since then I'm seeing multiple signal missed issues.To resolve, I've got to restart the collector, one time had to disconnect Bluetooth, then reconnect. Before the update I rarely had a signal missed issue. Have rolled back to 2021.03.08 and it worked again. Tested by upgrading to each release from 8th March, and found they work up to 2021.05.21. After this release, starting with 3b8665e-2021.06.11 missed signals with happen within 90 minutes of installing. 

Phone brand and model: Samsung Galaxy A70
Android version on the phone: 11 One UI 3.1
Sensor brand and model: Libre Freestyle 14 day US
Hardware data source: MioaMioa 2, OOPalgorithm fa218ef-de-libre2-sensors-2019.11.22 build 1911221114
Works fine with xDrip version: 3df0152-2021.05.21
Fails with the next xDrip version: 3b8665e-2021.06.11
![FB_IMG_1636584896525](https://user-images.githubusercontent.com/62366031/141207395-b89c64dd-d25f-46e0-aa24-0fb49610245f.jpg)
![FB_IMG_1636584902262](https://user-images.githubusercontent.com/62366031/141207419-54a1700b-cf05-4d58-b646-8e62f3fcf546.jpg)
![FB_IMG_1636584929000](https://user-images.githubusercontent.com/62366031/141207426-b5051abf-6fcf-46f0-a924-eb7de893620c.jpg)
![IMG-20211110-WA0004](https://user-images.githubusercontent.com/62366031/141207494-cd548ad6-9fa1-48f2-ae98-5ad2ad653481.jpg)

**Comment by Navid200** [comment]
This was brought up in the facebook group.  I cannot address it as it is Libre.  So, I suggested an issue to be opened.

**Comment by LuisPuertollano** [comment]
Hello people, :-)

I updated to last solid version (version 2021.10.04) 2 to 3 days ago and today I had the same problem.
If it helps, I can confirm that thurning back to xDrip version: 3df0152-2021.05.21 solves the issue.
Phone brand and model: Samsung Galaxy A52 5G
Android version on the phone: 11 One UI 3.1
Samsung Security Patch: October 2021
Sensor brand and model: Libre Freestyle 2
Hardware data source: Libre 2 (Patched App)

Thank you all! :-D
BR

**Comment by tzachi-dar** [comment]
There are two different issues here:
1) Us sensors, using mm2 and oop1
2) Libre2 sensors with patched app.

I'm not so familiar with the patched app so LuisPuertollano can you try installing oop2?

As for bjblackmore - Since the issue happens after 90 minutes, I suspect that something has changed in the way that our BT is working. But comparing the two revisions, I did not see a change in our BT code.

So, both: Can you run with the phone connected to a computer and send me the logs that you get from "adb logcat -v" ?
If not, can you follow the instructions at [minimal looper](https://www.minimallooper.com/post/how-to-setup-freestyle-libre-2-and-oop2-to-use-a-native-bluetooth-connection-in-xdrip) to add the extra tags for logging?
Then please send the logs to tzachi.dar@gmail.com

**Comment by bjblackmore** [comment]
Once I've enabled logging, do you want me to perform an upgrade to a non working xDrip version? Or do you only want the logs for the working version?

**Comment by Navid200** [comment]
@tzachi-dar I only asked @bjblackmore to open this issue.
Would you please focus on his issue and ignore everyone else?
He has gone back and forth between two consecutive releases of xDrip with consistent behaviors for the two releases.

If anyone else experiences a similar issue, please be patient allowing the intended case to be resolved first before coming forward with your case.  
You can always open a discussion for your situation.  https://github.com/Navid200/xDrip/discussions
It is possible that what you are experiencing is related.  But, even a single different component is enough to throw off the intent of the thread as you can see above.

**Comment by tzachi-dar** [comment]
Please only send the logs for the non-working version.

**Comment by tzachi-dar** [comment]
Hi,

The logs supplied show that there is a problem but I could not find what it is.

Please install the following version:
https://drive.google.com/uc?id=1nLaouqpNbbvqnmn0gugXD-uaeE0JI0fK&export=download
It is based on the latest code, and has more debug information.

(you will have to remove your version of xdrip from the phone, which means that you will have to backup your settings and db and restore them after the new install).

Once this is done, please send me your updated DB.
you probably need to keep running until you see something like:


2021-11-12 03:43:59 jamorham nsemulator NSEmulator onReceiver: com.eveningoutpost.dexdrip.NS_EMULATOR
2021-11-12 03:43:59 jamorham nsemulator Receiving NSEmulator broadcast
2021-11-12 03:43:59 LibreOOPAlgorithm handleData called with {"oOPResultsArray":[{"currentBg":75,"currentTime":11708,"currentTrend":0,"historicBg":[{"bg":79,"quality":0,"time":11235},{"bg":92,"quality":0,"time":11250},{"bg":103,"quality":0,"time":11265},{"bg":92,"quality":0,"time":11280},{"bg":79,"quality":0,"time":11295},{"bg":80,"quality":0,"time":11310},{"bg":84,"quality":0,"time":11325},{"bg":84,"quality":0,"time":11340},{"bg":85,"quality":0,"time":11355},{"bg":85,"quality":0,"time":11370},{"bg":83,"quality":0,"time":11385},{"bg":83,"quality":0,"time":11400},{"bg":82,"quality":0,"time":11415},{"bg":86,"quality":0,"time":11430},{"bg":107,"quality":0,"time":11445},{"bg":137,"quality":0,"time":11460},{"bg":153,"quality":0,"time":11475},{"bg":159,"quality":0,"time":11490},{"bg":166,"quality":0,"time":11505},{"bg":175,"quality":0,"time":11520},{"bg":179,"quality":0,"time":11535},{"bg":173,"quality":0,"time":11550},{"bg":158,"quality":0,"time":11565},{"bg":142,"quality":0,"time":11580},{"bg":129,"quality":0,"time":11595},{"bg":118,"quality":0,"time":11610},{"bg":109,"quality":0,"time":11625},{"bg":100,"quality":0,"time":11640},{"bg":94,"quality":0,"time":11655},{"bg":87,"quality":0,"time":11670},{"bg":82,"quality":0,"time":11685},{"bg":0,"quality":1,"time":11700}],"serialNumber":"","timestamp":1636681437492}],"version":1}

As for the other issue. Please talk with Navid as it probably requires a new issue to be opened.

If you can get the data using "adb logcat" this would really help with debugging that kind of issue.

**Comment by Navid200** [comment]
@bjblackmore This explains how to backup the database:
https://github.com/Navid200/xDrip/wiki/Backup-Database
You should do that right before when you intend to uninstall xDrip.
After, you should install the other version.
After the new version has been installed, you can follow this to restore the saved database:
https://github.com/Navid200/xDrip/wiki/Restore-Database

You can do the exact same thing every time you want to change xDrip to a version that cannot be installed over.

All the official xDrip versions have the same signature and you will not need to uninstall to change releases.
But, you have been asked to install a version released for troubleshooting, which has a different signature.

**Comment by tzachi-dar** [comment]
@bjblackmore Still waiting for data from you.

**Comment by Navid200** [comment]
@bjblackmore It's been almost 5 months since a developer attempted to help you get this issue resolved.
I hope that the reason you have ignored the request is that your problem has been resolved.  So, I am closing this issue.
We can always reopen if needed.
